internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -780,31 +794,39 @@ func (c *Coordinator) handleCoordinatorDone(ctx context.Context, varsErrCh, runt
 				timeouts = append(timeouts, "no response from varsWatcher component")
 			}
 			c.logger.Debugf("timeout while waiting for other components to shut down: %v", timeouts)
-			break
+			doneWait = true
 		case err := <-runtimeErrCh:
 			runtimeErr = err
+			returnedRuntime = true
```

Comment:
If these channels get closed, won't you discard any saved error? You could guard against this or just detect that the channel was closed with `case err, ok := <-runtimeErrCh:`